tests/core/test_expectation_suite.py

```diff
@@ -464,10 +465,46 @@ def _test_expectation_can_be_saved_after_update(self, context, expectation):
         updated_column_name = "foo"
         expectation.column = updated_column_name
         expectation.save()
+        assert suite.expectations[0].column == updated_column_name
         suite = context.get_expectation_suite(suite_name)
         assert len(suite.expectations) == 1
         assert suite.expectations[0].column == updated_column_name
 
+    @pytest.mark.filesystem
+    def test_expectation_save_only_persists_single_change(self, empty_data_context):
+        context = empty_data_context
+        suite_name = "test-suite"
+        expectations = [
+            ExpectColumnValuesToBeInSet(
+                column="a",
+                value_set=[1, 2, 3],
+            ),
+            ExpectColumnValuesToBeInSet(
+                column="b",
+                value_set=[4, 5, 6],
+            ),
+        ]
+
+        suite = context.add_expectation_suite(
+            expectation_suite_name=suite_name,
+            expectations=[e.configuration for e in expectations],
+        )
+
+        assert suite.expectations[0].column == "a"
+        assert suite.expectations[1].column == "b"
+
+        # Change the column names of both expectations
+        expectation = suite.expectations[0]
+        other_expectation = suite.expectations[1]
+        expectation.column = "c"
+        other_expectation.column = "d"
+
+        # Only persist change made to first expectation
+        expectation.save()
+
+        assert suite.expectations[0].column == "c"
+        assert suite.expectations[1].column == "b"
```

Comment:
Thanks for adding this test! just wanted to flag that it is actually not testing behavior as desired. `other_expectation.column` should still be `d` here, since the user made that change locally. Better would be to fetch the suite from the backend, and assert that the expectation in the remote suite hasn't been changed. I'll update in a followup PR.